### PR TITLE
Onclick

### PR DESCRIPTION
**Pull request by Nour-Cheour10**

### CODE REVIEW

File: Untitled.ipynb
Comment (martinRenou) [inline Untitled.ipynb:1]: This may have been commited by mistake

File: ipyopenlayers/Map.py [hunk 1]
```diff
@@ -166,4 +170,73 @@ def remove_control(self, control):
...
+    def _handle_click(self, lat, lon, api_key):
+        """
+        Handle click events and trigger registered callbacks.
+        """
+        print(f"Handling click event at Longitude: {lon}, Latitude: {lat}")
+        country = self.get_country_from_coordinates_geoapify(lat, lon, api_key)
+        self.clicked_country = country
+        print(f"Country: {country}")
+
+        callbacks = self._click_callbacks.get_callbacks() if self._click_callbacks else []
```
Comment (martinRenou) [inline ipyopenlayers/Map.py]: `_click_callbacks` is not an attribute of your class, you forgot to define it?

File: ipyopenlayers/Map.py [hunk 2]
```diff
@@ -166,4 +170,73 @@ def remove_control(self, control):
...
+            data = content.get('data', {})
+            method = data.get('method', '')
+
+            if method == 'custom':
+                event_data = data.get('content', {})
+                lon = event_data.get('lon')
+                lat = event_data.get('lat')
+                
+                if callable(self._click_callback):
+                    self._click_callback(lat, lon, self.get_country_from_coordinates_geoapify(lat, lon, api_key))
```
Comment (martinRenou) [inline ipyopenlayers/Map.py]: Do we really need to create a new `_click_callback` property here?

```suggestion
        def handle_frontend_event(widget, content, buffers):
            """Handle the click event from the frontend."""
            data = content.get('data', {})
            method = data.get('method', '')

            if method == 'custom':
                event_data = data.get('content', {})
                lon = event_data.get('lon')
                lat = event_data.get('lat')
                
                callback(lat, lon, self.get_country_from_coordinates_geoapify(lat, lon, api_key))
```

**Review by martinRenou — CHANGES_REQUESTED** [review]

**Comment by Nour-Cheour10** [comment]
please update snapshots

File: src/widget.ts [hunk 3]
```diff
@@ -143,6 +148,14 @@ export class MapView extends DOMWidgetView {
     this.model.on('change:zoom', this.zoomChanged, this);
     this.model.on('change:center', this.centerChanged, this);
   }
+
+  handleMapClick(event: MapBrowserEvent<MouseEvent>) {
+    const coordinate = event.coordinate;
+    const [lon, lat] = coordinate;
+    this.model.set('clicked_position', [lon, lat]);
+    this.model.save_changes();
```
Comment (martinRenou) [inline src/widget.ts]: I believe you don't need this in your implementation? You're not using it Python side

File: ipyopenlayers/openlayers.py [hunk 4]
```diff
@@ -259,96 +135,91 @@ class Map(DOMWidget):
...
-
-        Parameters
-        ----------
-        layer: Layer instance
-            The layer to remove.
-        """
         self.layers = [x for x in self.layers if x != layer]
     
     def remove_overlay(self, overlay):
-        """Remove an overlay from the map.
```
Comment (martinRenou) [inline ipyopenlayers/openlayers.py]: Your PR is removing the docstrings, maybe due to a wrong rebase

File: docs/source/_static/124.embed-bundle.js
Comment (martinRenou) [inline docs/source/_static/124.embed-bundle.js:1]: The content of this folder should not be commited

**Review by martinRenou — CHANGES_REQUESTED** [review]

**Comment by Nour-Cheour10** [comment]
please update snapshots

**Comment by Nour-Cheour10** [comment]
please update snapshots

File: ipyopenlayers/openlayers.py (hunk 4 above)
Comment (Nour-Cheour10) [reply ipyopenlayers/openlayers.py]: yes, fixed thanks

File: ipyopenlayers/openlayers.py [hunk 5]
```diff
@@ -351,4 +354,56 @@ def clear_layers(self):
...
+            self._msg_callback(self, content, buffers)
+            
+    def _handle_click(self, lat, lon):
+        """Handle click events and trigger registered callbacks."""
+
+
+        callbacks = self._click_callbacks.get_callbacks()
+        for callback in callbacks:
+            if callable(callback):
+                callback(lat, lon)
```
Comment (martinRenou) [inline ipyopenlayers/openlayers.py]:
```suggestion
```

**Comment by Nour-Cheour10** [comment]
please update snapshots

**Comment by Nour-Cheour10** [comment]
please update snapshots

File: .gitignore [hunk 6]
```diff
@@ -61,7 +61,7 @@ instance/
 
 # Sphinx documentation
 docs/_build/
-docs/_static/
+docs/source/_static/
```
Comment (martinRenou) [inline .gitignore]:
```suggestion
docs/_static/
```

Comment (Nour-Cheour10) [reply .gitignore]: this is  **docs/source/_static/** the path that I get when I build the project.

Comment (martinRenou) [reply .gitignore]: The docs are not located in `docs/source` anymore, are you sure you ran the docs build from the latest version?

Comment (Nour-Cheour10) [reply .gitignore]: yes,well the docs are not located in  `docs/source` but the files I get out of the build are
Screenshot from 2024-08-27 16-21-53

Comment (martinRenou) [reply .gitignore]: Ah I see what's wrong. We should fix it here: https://github.com/QuantStack/ipyopenlayers/blob/main/webpack.config.js#L78

Comment (Nour-Cheour10) [reply .gitignore]: okay thanks, I'll get it done.

**Comment by Nour-Cheour10** [comment]
please update snapshots

**Review by martinRenou — APPROVED** [review]
Thanks!

**Pull request merged**